Integrate new burner wallet flow

wryonik opened the pull request.

wryonik committed:
WIP: Integrate burner wallet flow

wryonik committed:
WIP: Integrate updated burner wallet flow

wryonik committed:
WIP: Integrate updated burner wallet flow

wryonik committed:
Lint fixes

wryonik committed:
Update node version to fix cache service unavailable error

wryonik committed:
Fix linting and prettier errors

wryonik committed:
Update timelock logic

JohnGuilding commented on src/components/burnerWallet/RequestedRecoveries.tsx:
```diff
@@ -113,38 +109,68 @@ const RequestedRecoveries = () => {
 
     // This function fetches the command template for the recoveryRequest API call. The command template will be in the following format: ['Recover', 'account', '{ethAddr}', 'using', 'recovery', 'hash', '{string}']
-    const command = (await readContract(config, {
+    // const command = (await readContract(config, {
```
Commented out code can be removed

JohnGuilding commented on src/components/burnerWallet/RequestedRecoveries.tsx:
```diff
@@ -264,15 +327,15 @@ const RequestedRecoveries = () => {
-  ) {
-    return <Loader />;
-  }
+  // // Since we are polling for every actions but only wants to show full screen loader for the initial request
```
Is this code important or can it be removed?

JohnGuilding commented on src/components/burnerWallet/RequestedRecoveries.tsx:
When the user executes recovery and there are enough approvals - is there a message that says "You can recover your account in 6 hours. This is a security feature to..."? Thinking that that could be useful UX in addition to the existing checks and error messages in the completeRecovery step

JohnGuilding reviewed (changes requested):
Thanks, great to get change made. Few comments

wryonik committed:
:chore: Update gitignore file to ignore yarn dependencies

wryonik committed:
:chore: Review changes

- Update contract addresses at one place
- Add a timer that tells users about the trigger recovery delay
- Remove commented out code
- Linting changes

wryonik committed:
chore: Copy changes

- Update "Gnosis Safe" -> "Safe"
- Remove "Under Audit" message
- Add explaination for burner wallet

wryonik committed:
:chore: Add checks on the complete recovery button

JohnGuilding commented:
Replaced by https://github.com/zkemail/email-recovery-ui/pull/13

JohnGuilding commented on src/components/burnerWallet/config.ts:
```diff
@@ -0,0 +1,39 @@
+  };
+};
+
+const config: Config = {
```
they're separate and no real need to consolidate them
